src/NodeApi/Runtime/NodeEmbeddingNodeApiScope.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.JavaScript.NodeApi.Runtime;
+
+using System;
+using static JSRuntime;
+using static NodejsRuntime;
+
+public sealed class NodeEmbeddingNodeApiScope : IDisposable
```

Comment:
Would it be possible / reasonable to define a new `JSValueScopeType` instead of this class? I know it's already confusing how many scope types there are, but another scope type might make more sense than this wrapper?

Reply:
It must be possible. Let me address this in a follow up PRs.